adjust defaults for prod best-practices and add dev mode (#1748)
apollographql/router: docs/source/configuration/overview.mdx

### Introspection

> By default, the router doesn't answer to introspection queries. You can override this behavior to enable the introspection like so:

Reviewer: ```suggestion
By default, the router does _not_ resolve introspection queries. You can enable introspection like so:
```

```yaml title="router.yaml"
#
# graphql: Configuration of the GraphQL server
#
graphql:
  introspection: true
```

### Landing page

> By default, the router doesn't display a landing page if you access its endpoint path via your browser. You can override this behavior to enable the landing page like so:

Reviewer: ```suggestion
By default, the router does _not_ display a landing page if you access its endpoint path via your browser. You can override this behavior to enable the landing page like so:
```

Reviewer: So it'll now display just _nothing_ by default? Just confirming that the language here is correct with the simple logic negation.

Reviewer 2: That's it
